tests/beta/test_default_qubit_tf.py

```diff
@@ -1045,174 +1020,195 @@ def circuit(a, b):
+        dev = qml.device("default.qubit.tf", wires=3, analytic=False, shots=1000)
 
-            if interface == "torch" and not torch_support:
-                pytest.skip("Skipped, no torch support")
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a, b):
+            qml.RX(a, wires=[0])
+            qml.RX(b, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
 
-            @qml.qnode(dev, diff_method=diff_method, interface=interface)
-            def circuit(x, w=None):
-                qml.RZ(x, wires=w)
-                return qml.expval(qml.PauliX(w))
+        a = tf.Variable(0.543)
+        b = tf.Variable(0.43)
 
-            return circuit(params[0], w=0)
+        res = circuit(a, b)
+        assert isinstance(res, tf.Tensor)
 
-        with pytest.raises(
-            ValueError,
-            match="Device default.qubit.tf only supports the tf interface when diff_method='backprop'",
-        ):
-            res = cost_raising_error(params)
+        # We don't check the expected value due to stochasticity, but
+        # leave it here for completeness.
+        # expected = [tf.cos(a), tf.cos(a) * tf.cos(b)]
+        # assert np.allclose(res, expected, atol=tol, rtol=0)
 
-    def test_error_classical_diff_autograd(self, tol):
-        """Tests that an error is raised if for the classical differentiation
-        method when using the autograd interface"""
-        interface = "autograd"
-        diff_method = "backprop"
+    def test_estimating_expectation_values_not_differentiable(self, tol):
+        """Test that analytic=False results in non-differentiable QNodes"""
+        dev = qml.device("default.qubit.tf", wires=3, analytic=False)
 
-        params = self.p
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a, b):
+            qml.RX(a, wires=[0])
+            qml.RX(b, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
 
-        def cost_raising_error(params):
-            # Cost within the test case such that the error can be caught
-            dev = qml.device("default.qubit.tf", wires=1)
+        a = tf.Variable(0.543)
+        b = tf.Variable(0.43)
 
-            @qml.qnode(dev, diff_method=diff_method, interface=interface)
-            def circuit(x, w=None):
-                qml.RZ(x, wires=w)
-                return qml.expval(qml.PauliX(w))
+        with tf.GradientTape() as tape:
+            res = circuit(a, b)
 
-            return circuit(params[0], w=0)
+        grad = tape.gradient(res, [a, b])
+        assert grad == [None, None]
 
-        with pytest.raises(
-            ValueError,
-            match="Device default.qubit.tf only supports the tf interface when diff_method='backprop'",
-        ):
-            res = cost_raising_error(params)
+    def test_qnode_collection_integration(self):

```

Comment:
This one doesn't seem to fit well in this test class

Reply:
I've split this out into its own class:
```python
class TestHighLevelIntegration:
    """Tests for integration with higher level components of PennyLane."""
```